DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,248 @@
+
+## Setting up your local environment
+
+In order to leverage local development tooling (including a rich live-preview capability that exists within the Docusaurus framework), you will
+need to ensure that you have a compatible version of Node.js installed on your system.
+
+### Node.js setup
+
+The easiest way to get started with Node is to use the [Node Version Manager (nvm)](https://github.com/nvm-sh/nvm). `nvm` will allow you to seamlessly manage
+different deployments of Node.js, offering simple command line tool access to switch between versions.
+
+> Note: on OSX, you will need to have the Xcode command line tools installed before installing `nvm`. These can be directly installed from [Apple Developer Resources](https://developer.apple.com/xcode/resources/).
+
+Installation of `nvm` can be done directly via a `curl` (alternatively, you can use `wget`) command:
+
+```bash
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+```
+
+Once nvm is installed, you can download and install a version of node that is supported with Docusaurus.
+
+```bash
+nvm ls
+```
+
+An example output of running the listing command is:
+
+```bash
+➜ nvm ls
+->     v16.20.2
+       v20.11.0
+         system
+default -> 16 (-> v16.20.2)
+iojs -> N/A (default)
+unstable -> N/A (default)
+node -> stable (-> v20.11.0) (default)
+stable -> 20.11 (-> v20.11.0) (default)
+lts/* -> lts/iron (-> v20.11.0)
+lts/argon -> v4.9.1 (-> N/A)
+lts/boron -> v6.17.1 (-> N/A)
+lts/carbon -> v8.17.0 (-> N/A)
+lts/dubnium -> v10.24.1 (-> N/A)
+lts/erbium -> v12.22.12 (-> N/A)
+lts/fermium -> v14.21.3 (-> N/A)
+lts/gallium -> v16.20.2
+lts/hydrogen -> v18.19.0 (-> N/A)
+lts/iron -> v20.11.0
+```
+
+In order to download a version of node using nvm that meets the `node>=18` requirements, you can:
+
+```bash
+nvm install lts/iron
+```
+This will fetch, per the list above, node `v20.12.0`.
+
+```bash
+➜ nvm install lts/iron
+Downloading and installing node v20.12.0...
+Downloading https://nodejs.org/dist/v20.12.0/node-v20.12.0-darwin-arm64.tar.xz...
+############################################################################################################################################################ 100.0%
+Computing checksum with sha256sum
+Checksums matched!
+Now using node v20.12.0 (npm v10.5.0)
+(mlflow-dev-env)
+```
+
+Once you have node installed, you can activate this new version by running:
+
+```bash
+➜ nvm use v20.12.0
+Now using node v20.12.0 (npm v10.5.0)
+```
+
+### Yarn setup
+
+With the above steps completed, nvm will have acquired both node and `npm` (The Node package manager).
+
+We'll use `npm` to install `yarn` so that we can do active development, build a static version of the website, and compile the blog markdown content into renderable `.tsx` files.
+
+```bash
+npm install --global yarn
+```
+
```

Comment:
The same comment for yarn.

https://github.com/mlflow/mlflow-website/pull/48/files#r1545880010